src/client/store/pages/assessmentSection/slice.ts

```diff
@@ -37,8 +37,19 @@ export const assessmentSectionSlice = createSlice({
       const { countryIso, nodeUpdate } = payload
       const { tableName, variableName, colName, value } = nodeUpdate
       const data = (state.data ?? {}) as TableData
-      const dataUpdate = TableDatas.updateDatum({ data, countryIso, tableName, variableName, colName, value })
-      state.data = { ...data, ...dataUpdate }
+      const dataUpdateProps = { data, countryIso, tableName, variableName, colName, value }
+      if (value.odp) {
+        // If the value overrides ODP value, update the ODP state
+        const dataUpdate = TableDatas.updateDatum({
+          ...dataUpdateProps,
+          data: state.originalDataPointData,
+          tableName: 'originalDataPointValue',
+        })
+        state.originalDataPointData = { ...state.originalDataPointData, ...dataUpdate }
+      } else {
+        const dataUpdate = TableDatas.updateDatum(dataUpdateProps)
+        state.data = { ...data, ...dataUpdate }
+      }
```

Comment:
When we update->receive an ODP value from websocket, handle it here

Although, when writing this and working with the websockets, I was wondering should we just use one state variable (eg. `data` here) and receive the data directly from backend?

Do we ever use non-odp values for odp years when 1b has ODPs enabled?

Reply:
I dont understand what's happening here 😅we'll talk in person

Reply:
yes